keras/preprocessing/image.py

```diff
@@ -248,9 +248,12 @@ def reset(self):
         self.batch_index = 0
 
     def _flow_index(self, N, batch_size=32, shuffle=False, seed=None):
+        # ensure self.batch_index is 0
+        self.reset()
```

Comment:
This seems to make `reset()` in `flow()` superfluous.  If so I think it would be cleaner and still safe to just remove the `reset()` from flow and have the index_array generator handle resetting all its counters.